crates/lair_keystore_api/src/lair_server.rs

```diff
@@ -0,0 +1,101 @@
+//! host a lair keystore
+
+use crate::lair_api::traits::AsLairCodec;
+use crate::prelude::*;
+use futures::future::{BoxFuture, FutureExt};
+use futures::stream::StreamExt;
+use parking_lot::RwLock;
+use std::future::Future;
+use std::sync::atomic;
+use std::sync::Arc;
+
+/// Traits related to LairServer. Unless you're writing a new
+/// implementation, you probably don't need these.
+pub mod traits {
+    use super::*;
+
+    /// trait object type for AsyncWrite instance.
+    pub type RawSend = Box<dyn tokio::io::AsyncWrite + 'static + Send + Unpin>;
+
+    /// trait object type for AsyncRead instance.
+    pub type RawRecv = Box<dyn tokio::io::AsyncRead + 'static + Send + Unpin>;
+
+    /// host a lair keystore
+    pub trait AsLairServer: 'static + Send + Sync {
+        /// It is much more secure to unlock a lair keystore server
```

Comment:
more secure than what?

Reply:
than having a client unlock it over a connection, but this whole locked concept is deleted in https://github.com/holochain/lair/pull/71